### ISSUE TITLE
[Bug]: [laion1b-test] mixcoord panic: invalid memory address or nil pointer dereference when loading segment

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#29575]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: cardinal-milvus-io-2.3-07ef52e-20231228
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):   pulsar 
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

mixcoord panic when reloading collection:
[mix_68rb6.log](https://github.com/milvus-io/milvus/files/13786853/mix_68rb6.log)

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4380571]

goroutine 2950389 [running]:
panic({0x49e6980, 0x728c820})
    /usr/local/go/src/runtime/panic.go:987 +0x3bb fp=0xc01084d6d0 sp=0xc01084d610 pc=0x1a25c5b
runtime.panicmem(...)
    /usr/local/go/src/runtime/panic.go:260
runtime.sigpanic()
    /usr/local/go/src/runtime/signal_unix.go:841 +0x37d fp=0xc01084d730 sp=0xc01084d6d0 pc=0x1a3de9d
github.com/milvus-io/milvus/internal/querycoordv2/task.(*Executor).loadSegment(0xc058b9e7e0, 0xc04624ee80, 0x0?)
    /go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/executor.go:199 +0xe91 fp=0xc01084df50 sp=0xc01084d730 pc=0x4380571
github.com/milvus-io/milvus/internal/querycoordv2/task.(*Executor).executeSegmentAction(0xc02efa0240?, 0xc04624ee80, 0x0)
    /go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/executor.go:132 +0x73 fp=0xc01084df80 sp=0xc01084df50 pc=0x437f673
github.com/milvus-io/milvus/internal/querycoordv2/task.(*Executor).Execute.func1()
    /go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/executor.go:107 +0xc5 fp=0xc01084dfe0 sp=0xc01084df80 pc=0x437f185
runtime.goexit()
    /usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc01084dfe8 sp=0xc01084dfe0 pc=0x1a5f9e1
created by github.com/milvus-io/milvus/internal/querycoordv2/task.(*Executor).Execute
    /go/src/github.com/milvus-io/milvus/internal/querycoordv2/task/executor.go:103 +0x5a5
``` 

### Steps To Reproduce

```markdown
argo: [laion1b-test-read-write-6]https://argo-workflows.zilliz.cc/archived-workflows/qa/2c8d015e-4eab-41fd-9226-cb8c8991140b?nodeId=laion1b-test-read-write-6
```

**Comment by congqixia** [comment]
/assign @ThreadDao 
patch merged ,FYI

**Comment by ThreadDao** [comment]
fixed `2.3-20231229-7a192da8-amd64`
